lib/server.js

```diff
@@ -334,6 +334,50 @@ async function create (log, error, config, routes, db, oauthdb, translator) {
   await server.register(require('hapi-fxa-oauth'))
 
   server.auth.strategy('oauthToken', 'fxa-oauth', config.oauth)
+
+  server.auth.scheme('fxa-oauth-refreshToken', (server, options) => {
+    return {
+      async authenticate (request, h) {
+        try {
+          const refreshTokenId = request.headers.authorization.split(' ')[1];
+          // TODO: verify that first chunk is Bearer?
+          const refreshToken = await oauthdb.checkRefreshToken(refreshTokenId)
+          if (! refreshToken || ! refreshToken.active) {
+            return h.unauthenticated()
+          }
+
+          const credentials = {
+            uid: refreshToken.sub,
+            refreshTokenId: refreshTokenId,
```

Comment:
So, we want to be a bit careful here. The oauth-server stores refresh tokens *hashes* in the db rather than raw tokens, and we should probably do the same here.  You could either hash it explicitly like you do here:

  https://github.com/mozilla/fxa-auth-server/pull/2937/files#diff-ef91ae14c960c9bb4f51ad0b805f685eR41

Or you could use the value you're returning as the `jti` field here:

  https://github.com/mozilla/fxa-auth-server/pull/2937/files#diff-ef91ae14c960c9bb4f51ad0b805f685eR72

But we definitely don't want to write the raw refreshToken into the database.